### ISSUE TITLE
FDSN: misleading error message when failing to read invalid FDSN StationXML response (garbled xml)

### ISSUE THREAD

**Issue by filefolder** [obspy/obspy#3038]
**### FYI the below now works (xml was garbled due to improper handling of karelian alphabet characters which is a separate issue) ###**

There seems to be some sort of formatting glitch when requesting a fairly elaborate array of channels via get_stations. I know this is sort of a silly way to do it, but it should still work?

The below work as expected:

```
inv = ap.get_stations(network='*',level='channel', channel='CH*')
inv = ap.get_stations(network='*',level='channel', channel='HH*')
inv = ap.get_stations(network='*',level='channel', channel='BH*')
inv = ap.get_stations(network='*',level='channel', channel='EH*')
inv = ap.get_stations(network='*',level='channel', channel='SH*')
```

and some combinations work

```
inv = ap.get_stations(network='*',level='channel', channel='BH*,CH*,SH*')
inv = ap.get_stations(network='*',level='channel', channel='CH*,SH*,EH*,HH*')
```

but when I try to do all five at once, 

```
inv = ap.get_stations(network='*',level='channel', channel='BH*,CH*,SH*,EH*,HH*')

~/.local/lib/python3.8/site-packages/obspy/clients/fdsn/client.py in get_stations(self, starttime, endtime, startbefore, startafter, endbefore, endafter, network, station, location, channel, minlatitude, maxlatitude, minlongitude, maxlongitude, latitude, longitude, minradius, maxradius, level, includerestricted, includeavailability, updatedafter, matchtimeseries, filename, format, **kwargs)
    751         else:
    752             # This works with XML and StationXML data.
--> 753             inventory = read_inventory(data_stream)
    754             data_stream.close()
    755             return inventory

<decorator-gen-157> in read_inventory(path_or_file_object, format, level, *args, **kwargs)

~/.local/lib/python3.8/site-packages/obspy/core/util/decorator.py in _map_example_filename(func, *args, **kwargs)
    289                         except IOError:
    290                             pass
--> 291         return func(*args, **kwargs)
    292     return _map_example_filename
    293 

~/.local/lib/python3.8/site-packages/obspy/core/inventory/inventory.py in read_inventory(path_or_file_object, format, level, *args, **kwargs)
     89         return _create_example_inventory()
     90     else:
---> 91         return _generic_reader(path_or_file_object, _read, format=format,
     92                                **kwargs)
     93 

~/.local/lib/python3.8/site-packages/obspy/core/util/base.py in _generic_reader(pathname_or_url, callback_func, **kwargs)
    628             with NamedTemporaryFile() as fh:
    629                 fh.write(pathname_or_url.read())
--> 630                 generic = callback_func(fh.name, **kwargs)
    631         return generic
    632     elif isinstance(pathname_or_url, bytes) and \

<decorator-gen-158> in _read(filename, format, **kwargs)

~/.local/lib/python3.8/site-packages/obspy/core/util/decorator.py in uncompress_file(func, filename, *args, **kwargs)
    200     else:
    201         # no compressions
--> 202         result = func(filename, *args, **kwargs)
    203     return result
    204 

~/.local/lib/python3.8/site-packages/obspy/core/inventory/inventory.py in _read(filename, format, **kwargs)
     98     Reads a single event file into a ObsPy Catalog object.
     99     """
--> 100     inventory, format = _read_from_plugin('inventory', filename, format=format,
    101                                           **kwargs)
    102     return inventory

~/.local/lib/python3.8/site-packages/obspy/core/util/base.py in _read_from_plugin(plugin_type, filename, format, **kwargs)
    400                 break
    401         else:
--> 402             raise TypeError('Unknown format for file %s' % filename)
    403     else:
    404         # format given via argument

TypeError: Unknown format for file /tmp/obspy-vic97a1s.tmp
```

I thought there may be a limit of 5 channels but I can change one out and it works still

`inv = ap.get_stations(network='*',level='channel', channel='BH*,CH*,SH*,EH*,DH*')`

Also just selecting EVERYTHING works

`inv = ap.get_stations(network='*',level='channel', channel='*')`

Not critical, just peculiar. I am not sure how to intercept the tmp file before it is destroyed so can't take a peek at what's actually going wrong. Haven't tried other networks but I don't think it should be different?

### Code to Reproduce

```python
from obspy.clients.fdsn import Client
ap = Client('AUSPASS')
inv = ap.get_stations(network='*', level='channel', channel='CH*,BH*,HH*,SH*,EH*')
```

### ObsPy Version?

1.3

### Operating System?

Ubuntu 20.04

### Python Version?

3.8.10

### Installation Method?

pip

**Comment by megies** [comment]
Turns out this isn't a problem of the FDSN client, there is some problem reading the response from the server, we fail to read it as StationXML:

```python
import io
import requests
from obspy import read_inventory

url = 'http://auspass.edu.au/fdsnws/station/1/query?network=%2A&channel=CH%2A%2CBH%2A%2CHH%2A%2CSH%2A%2CEH%2A&level=channel'
result = requests.get(url)
with io.BytesIO(result.content) as bio:
    inv = read_inventory(bio, 'STATIONXML')
```


That said, the error you get there indeed is misleading, we should ideally explicitly raise a proper error message after failing to read what the server returned.

fdsn/client.py
```python
 743         url = self._create_url_from_parameters(
 744             "station", DEFAULT_PARAMETERS['station'], kwargs)
 745                                                                                     
 746         data_stream = self._download(url)                                           
 747         data_stream.seek(0, 0)                                                      
 748         if filename:                                                                
 749             self._write_to_file_object(filename, data_stream)                       
 750             data_stream.close()                                                     
 751         else:                                                                       
 752             # This works with XML and StationXML data.                              
 753             inventory = read_inventory(data_stream)       # <------------                           
 754             data_stream.close()                                                     
 755             return inventory                                                        
 756                                                                                     
```
The line with the arrow shows that weird error after going through file format detection. In principle we shouldn't even need file format detection, it should be clear what we get from the `format` kwarg's value in `get_stations()`.

**Comment by megies** [comment]
```python
import requests
from obspy import read_inventory

url = 'http://auspass.edu.au/fdsnws/station/1/query?network=%2A&channel=CH%2A%2CBH%2A%2CHH%2A%2CSH%2A%2CEH%2A&level=channel'
result = requests.get(url)

with open('/tmp/output.xml', 'wb') as fh:
    fh.write(result.content)

print(read_inventory('/tmp/output.xml', 'STATIONXML'))
```

```
  File "/home/megies/git/obspy/obspy/io/stationxml/core.py", line 152, in _read_stationxml
    root = etree.parse(path_or_file_object).getroot()
  File "src/lxml/etree.pyx", line 3536, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1876, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1902, in lxml.etree._parseDocumentFromURL
  File "src/lxml/parser.pxi", line 1805, in lxml.etree._parseDocFromFile
  File "src/lxml/parser.pxi", line 1177, in lxml.etree._BaseParser._parseDocFromFile
  File "src/lxml/parser.pxi", line 615, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 725, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 654, in lxml.etree._raiseParseError
  File "/tmp/output.xml", line 2
lxml.etree.XMLSyntaxError: Opening and ending tag mismatch: Value line 2 and Site, line 2, column 1041095
```

**Comment by megies** [comment]
Couldn't find real quick what the issue with the xml is but it seems like a garbled xml response to me at this point..

**Comment by filefolder** [comment]
Thanks-- I have a suspicion there is some internal anti-DDOS throttling going via our IT dept when too many small requests are placed at once. I suppose this happens when listing the channels as such vs just clobbering for them all simultaneously. 

It may be worth it to change the error message but on the other hand it's a pretty niche issue. I'll close anyway

**Comment by megies** [comment]
Nevermind, found it..

```
<Network code="7S" startDate="2006-10-02T23:17:59" endDate="2007-08-26T06:00:01.24" restrictedStatus="open"><Description>SETA / SE Tasmania Array</Description><Comment id="0"><Value>https://www.fdsn.org/networks/detail/7S_2006/</Value></Comment><Comment id="1"><Value>N. Rawlinson, H. Tkal?<Country>Australia</Country></Site><CreationDate>2006-10-05T05:24:59</CreationDate>
```

That comment tag is garbled. I assume it's a problem of the FDSN web service implementation, hitting something in the comment that puts the parser off and then simply continuing with the following tags, or something of the sort.

**Comment by megies** [comment]
> Thanks-- I have a suspicion there is some internal anti-DDOS throttling going via our IT dept when too many small requests are placed at once. I suppose this happens when listing the channels as such vs just clobbering for them all simultaneously.
> 
> It may be worth it to change the error message but on the other hand it's a pretty niche issue. I'll close anyway

Nah nothing of the like, you wanna probably debug the FDSN web service implementation, see above.

**Comment by filefolder** [comment]
Ah I see, yes it is a character in the name Tkalčić. This is a whole other ball of wax that I have been dealing with with certain inventory objects. 

Very strange that it only occurs when requesting inventory in this particular way, but not when requesting it more directly.

**Comment by megies** [comment]
I believe this can be closed now that #3212 is merged, reopen if not
